woocommerce/woocommerce
# Add remote spec rule validator UI

### Changes proposed in this Pull Request:





This PR adds `Remote Spec Rule Validator` in WooCommerce Admin Test Helper to help testing remote spec rules.




### How to test the changes in this Pull Request:

1. Checkout this branch.
2. Run `pnpm --filter=@woocommerce/plugin-woocommerce-beta-tester start` to build the latest changes.
3. If you haven't already, symlink `woocommerce-beta-tester` directory to your plugins dir. 
4. Activate `woocommerce-beta-tester` plugin.
5. Go to `Tools -> WCA Test Helper` and click on `Remote Spec Rule Validator` tab.
6. The textarea already has a sample rule.
7. Click on `Validate` button.
8. Confirm the rule passes.
9. Test with other rules from WCCOM feeds (e.g, https://woocommerce.com/wp-json/wccom/payment-gateway-suggestions/1.0/suggestions.json) 



![Screen Shot 2024-02-23 at 3 30 49 PM](https://github.com/woocommerce/woocommerce/assets/4723145/8201a645-96f6-4538-a32d-351edd3097da)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
